### PR TITLE
fix(auth0): made AuthenticationClient members (e.g. oauth) always defined

### PR DESCRIPTION
**Pull request by GauBen**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] [Add or edit tests](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#my-package-teststs) to reflect the change.
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] [Run `npm test <package to test>`](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#running-tests).

If changing an existing definition:
- [x] Provide a URL to documentation or source code which provides context for the suggested changes: https://github.com/auth0/node-auth0/blob/bd7cfc31a45ca68a549445a44b102998193efe88/src/auth/index.js#L82-L115
- [ ] If this PR brings the type definitions up to date with a new version of the JS library, update the version number in the header.

These 5 members are set in the constructor

### CODE REVIEW

**Comment by GauBen** [comment]
I'm not sure I made the right things but I updated tsconfig to match https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#tsconfigjson

**Comment by GauBen** [comment]
I'm updating the tests rather than the type definition

File: types/auth0/index.d.ts [hunk 1]
```diff
@@ -1157,11 +1157,11 @@ export interface GrantResponse {
...
-    database?: DatabaseAuthenticator | undefined;
-    oauth?: OAuthAuthenticator | undefined;
-    passwordless?: PasswordlessAuthenticator | undefined;
-    tokens?: TokensManager | undefined;
-    users?: UsersManager | undefined;
+    database: DatabaseAuthenticator;
+    oauth: OAuthAuthenticator;
+    passwordless: PasswordlessAuthenticator;
+    tokens: TokensManager;
+    users: UsersManager;
```
Comment (peterblazejewicz) [inline types/auth0/index.d.ts:1164]: so those also looks like read-only?, if not available to be set (we've just leaked details in types for no obvious gain for users)

**Review by gabritto — APPROVED** [review]

**Comment by GauBen** [comment]
Ready to merge

**Pull request merged**